tests/test_logging.js

```diff
@@ -23,12 +23,13 @@ test('match debugging works', async t => {
   const exampleBody = 'Hello yourself!'
   await got.post('http://example.test/deep/link', { body: exampleBody })
 
-  t.ok(log.calledOnce)
-  t.equal(
-    JSON.parse(log.getCall(0).args[1]).href,
-    'http://example.test/deep/link'
+  t.ok(
+    log.calledWith(
+      sinon.match.string,
+      sinon.match('http://example.test/deep/link'),
+      sinon.match(exampleBody)
+    )
   )
-  t.equal(JSON.parse(log.getCall(0).args[2]), exampleBody)
```

Comment:
This test broke because I moved a could debug statements. It seems odd to me to test such a thing, but 🤷‍♂.
I updated this test to verify that the one debug was still be called when a match occurs.

Reply:
Gotcha. Yea, I think I added this test. In my experience it is easy for logging code to be broken. Two ways I've seen it fail is that conditional logging code crashes, and I've often seen code that logs entirely the wrong thing, like a templated string instead of the filled in string. The test makes it possible to refactor the logging with confidence.